src/sql/parts/modelComponents/componentBase.ts

```diff
@@ -46,7 +46,9 @@ export abstract class ComponentBase extends Disposable implements IComponent, On
 	protected _onEventEmitter = new Emitter<IComponentEventArgs>();
 
 	public layout(): void {
-		this._changeRef.detectChanges();
+		if (!this._changeRef['destroyed']) {
```

Comment:
Nice change. We've verified all places this is used?

Reply:
I verified all the component calling the parent layout method. not sure any other way to do it

---
In reply to: [214410826](https://github.com/Microsoft/sqlopsstudio/pull/2351#discussion_r214410826) [](ancestors = 214410826)